src/builder.rs

```diff
@@ -8,11 +8,25 @@ use std::process::Stdio;
 use std::str;
 use std::{fs, io};
 
-use dirs::state_dir;
 use once_cell::sync::OnceCell;
 use tempfile::{Builder, TempDir};
 use tokio::process;
 
+fn state_dir() -> Option<PathBuf> {
+    fn get_absolute_path(path: OsString) -> Option<PathBuf> {
+        let path = PathBuf::from(path);
+        path.is_absolute().then_some(path)
+    }
+
+    if let Some(xdg) = std::env::var_os("XDG_STATE_HOME") {
+        get_absolute_path(xdg)
+    } else if let Some(home) = std::env::var_os("HOME") {
```

Comment:
On unix, if env HOME is not set, you would need to call getpwuid_r to determin home.

Luckily, std already provides it, although it's deprecated due to impl problems on windows:

```suggestion
    } else if let Some(home) = std::env::home_dir() {
```

Reply:
Done, I also made sure it avoids calling `home_dir()` on Windows.